// Copyright 2011, Tencent Inc.
// Author: Hangjun Ye (hansye@tencent.com)

import "common/rpc/proto_rpc/rpc_option.proto";

package rpc_examples;

message EchoRequest {
  required string user = 1;
  required string message = 2;
}

message EchoResponse {
  required string user = 1;
  required string message = 2;
}

// The server side serivce.
service EchoServer {
  // The service timeout is 5 seconds.
  option (rpc.rpc_service_timeout) = 5000;

  rpc SimpleEcho(EchoRequest) returns(EchoResponse);

  rpc ComplicatedEcho(EchoRequest) returns(EchoResponse) {
    // This method specific timeout is 10 seconds.
    option (rpc.rpc_method_timeout) = 10000;
  }

  rpc TimeoutEcho(EchoRequest) returns(EchoResponse) {
    // This method specific timeout is 1 second, to test the handling of
    // request timeout.
    option (rpc.rpc_method_timeout) = 1000;
  }
}

// The dual echo service at server side, to test the "dual" rpc.
service DualEchoServer {
  rpc DualEcho(EchoRequest) returns(EchoResponse);
}

// The dual echo service at client side, to test the "dual" rpc.
service DualEchoClient {
  rpc CollectDualEcho(EchoRequest) returns(EchoResponse);
}
